cmake/Compiler/GNUARM.cmake

```diff
@@ -50,7 +50,5 @@ macro(__compiler_gnuarm lang)
     set(CMAKE_${lang}_CREATE_STATIC_LIBRARY  "<CMAKE_AR> rsc <TARGET> <LINK_FLAGS> <OBJECTS>")
     set(CMAKE_${lang}_CREATE_PREPROCESSED_SOURCE "<CMAKE_${lang}_COMPILER> <DEFINES> <INCLUDES> <FLAGS> -E <SOURCE> > <PREPROCESSED_SOURCE>")
     set(CMAKE_${lang}_CREATE_ASSEMBLY_SOURCE "<CMAKE_${lang}_COMPILER> <DEFINES> <INCLUDES> <FLAGS> -S <SOURCE> -o <ASSEMBLY_SOURCE>")
-
-    set(CMAKE_DEPFILE_FLAGS_${lang} "--depend=<DEPFILE> --depend_single_line --no_depend_system_headers")
```

Comment:
This is a common CMake built-in feature used for toolchains.
So we should be completely sure before removing.

But one thing I do notice, is that this line is not valid for `arm-none-eabi-gcc` cause it is a `armcc` extension.
http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0472m/chr1359124912122.html

Question is if this should be changed to the corresponding GNU extension:
```
    set(CMAKE_DEPFILE_FLAGS_${lang} "-MD -MT <OBJECT> -MF <DEPFILE>")  
```
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Compiler/GNU.cmake#L52

Reply:
I added this line, and it seems to work fine, thanks!